servers/audio_server.h

```diff
@@ -56,15 +56,26 @@ class AudioDriver {
 	SafeNumeric<uint64_t> prof_time;
 #endif
 
+public:
+	struct SizePosition {
+		unsigned int size;
+		unsigned int position;
+
+		SizePosition(unsigned int p_size = 0, unsigned int p_position = 0) noexcept :
+				size(p_size), position(p_position) {}
+	};
+
 protected:
-	Vector<int32_t> input_buffer;
-	unsigned int input_position = 0;
-	unsigned int input_size = 0;
+	LocalVector<int32_t> input_buffer;
+	std::atomic<SizePosition> input_read;
+	SizePosition input_write;
```

Comment:
`input_write` doesn't have to be `std::atomic`, mirrors behaviour from the previous implementation.
Copied from: https://github.com/godotengine/godot/pull/92969#discussion_r1634436636.